test/xpath/test_base.rb

```diff
@@ -416,12 +416,103 @@ def test_preceding
       assert_equal( 4, cs.length )
     end
 
-    def test_preceding_sibling
-      d = REXML::Document.new("<a><b><c/><d/><x/></b><b><e/><x/></b></a>")
-      matches = REXML::XPath.match(d, "a/b/x/preceding-sibling::node()")
+    def test_preceding_multiple
+      source = <<-XML
+<a>
+  <b/><c/><d/><d/><e/><f/>
+</a>
+      XML
+      d = REXML::Document.new( source )
```

Comment:
* Let's use `XXX(YYY)` style not `XXX( YYY )` style for newly added codes
* Let's use `doc` or `document` instead of `d` because we have `<d/>` in this context

```suggestion
      doc = REXML::Document.new(source)
```

Reply:
OK.
I see.